lua/cfc_disconnect_interface/client/cl_interface.lua

```diff
@@ -513,9 +518,74 @@ local function createInterface()
             end
         end
     end
+    -- Custom method used for hiding the main popup
+    function frame:OnHide()
+        local miniWindow = frame._miniWindow
+        if miniWindow then
+            miniWindow:Show()
+            local rejoinOn = buttonsPanel.reconBtn.autoJoin
+            miniWindow.rejoinLabel:setTextAndAlign( "Rejoin " .. ( rejoinOn and "enabled" or "disabled" ) )
+            miniWindow.rejoinLabel:SetTextColor( rejoinOn and greenCol or yellowCol )
+            return
+        end
+
+        miniWindow = vgui.Create( "DFrame" )
+        frame._miniWindow = miniWindow
+        local miniW, miniH = 200, 100
+        miniWindow:SetSize( miniW, miniH )
+        local screenWidth = ScrW()
+        miniWindow:SetPos( screenWidth - miniW - 30, 30 )
+        miniWindow:SetTitle( "Mini window" )
+        miniWindow:SetDraggable( true )
+        miniWindow:SetScreenLock( true )
+        miniWindow:ShowCloseButton( false )
+
+        local curTimeLabel = makeLabel( miniWindow, secondsAsTime( math.floor( timeDown ) ), 30, Color( 251, 191, 83 ), 0.5, "CFC_Mono" )
+        function curTimeLabel:Think()
+            if apiState ~= CFCCrashAPI.SERVER_UP then
+                self:setTextAndAlign( secondsAsTime( math.floor( timeDown ) ) )
+            else
+                self:SetTextColor( greenCol )
+            end
+        end
+
+        local btn = vgui.Create( "DButton", miniWindow )
+        btn:SetText( "" )
+        btn:SetSize( miniW, miniH - 25 )
+        btn:SetPos( 0, 25 )
+        btn:SetMouseInputEnabled( true )
+
+        function btn:Paint( w, h )
+            if not btn:IsHovered() then return end
+            surface.SetDrawColor( accentCol )
+            surface.DrawRect( 0, 0, w, h )
+
+            surface.SetFont( "CFC_Special" )
+            surface.SetTextColor( greenCol:Unpack() )
+            surface.SetTextPos( w / 4, 10 )
+            surface.DrawText( "Maximize" )
+        end
+        btn.Think = frame.Think
+        function btn:DoClick()
+            frame:Show()
+            miniWindow:Hide()
+        end
+
+        function miniWindow:Paint( w, h )
+            surface.SetDrawColor( primaryCol )
+            surface.DrawRect( 0, 0, w, h )
+        end
+
+        local rejoinOn = buttonsPanel.reconBtn.autoJoin
+        miniWindow.rejoinLabel = makeLabel( miniWindow, "Rejoin " .. ( rejoinOn and "enabled" or "disabled" ), 60, rejoinOn and greenCol or yellowCol )
```

Comment:
nit: This line has got a bit long, I'd prefer you split the conditions into their own lines, but I won't hold the PR up for it